play-services-core/src/main/kotlin/com/google/android/gms/potokens/utils/PoTokenHelper.kt

```diff
@@ -0,0 +1,233 @@
+package com.google.android.gms.potokens.utils
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.text.TextUtils
+import android.util.Base64
+import android.util.Log
+import com.android.volley.AuthFailureError
+import com.android.volley.NetworkResponse
+import com.android.volley.Request
+import com.android.volley.Response
+import com.android.volley.VolleyError
+import com.android.volley.toolbox.RequestFuture
+import com.android.volley.toolbox.Volley
+import com.google.android.gms.droidguard.DroidGuard
+import com.google.android.gms.droidguard.internal.DroidGuardResultsRequest
+import com.google.android.gms.potokens.CipherKey
+import com.google.android.gms.potokens.GetPoIntegrityTokenRequest
+import com.google.android.gms.potokens.GetPoIntegrityTokenResponse
+import com.google.android.gms.potokens.Key
+import com.google.android.gms.potokens.KeyData
+import com.google.android.gms.potokens.KeySet
+import com.google.android.gms.potokens.PoTokenInfo
+import com.google.android.gms.potokens.PoTokenResult
+import com.google.android.gms.potokens.PoTokenResultWrap
+import com.google.android.gms.potokens.internal.TAG
+import com.google.android.gms.tasks.Tasks
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import org.microg.gms.common.Constants
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.util.Random
+import java.util.concurrent.TimeUnit
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+import kotlin.math.abs
+
+class PoTokenHelper {
+
+    private val tokenUrl: String
+        get() = "https://deviceintegritytokens-pa.googleapis.com/v1/getPoIntegrityToken?alt=proto&key=AIzaSyBtL0AK6Hzgr69rQyeyhi-V1lmtsPGZd1M"
```

Comment:
I think that it is better if the part "AIzaS..." is moved to a constant.

Reply:
you are right, it has been moved to PoTokenConstants.kt